alchy/manager.py

```diff
@@ -115,11 +114,28 @@ class Manager(ManagerMixin):
     your subclass will inherit the functionality of :class:`alchy.Session`.
     """
 
-    def __init__(self,
-                 config=None,
-                 session_options=None,
-                 Model=None,
-                 session_class=None):
+    def __init__(self, config=None, session_options=None,
+                 Model=None, session_class=None):
+        self.config = None
+        self._engines = {}
+        self._binds = {}
+        self.session_class = None
+        self.session = None
+        self.Model = None
+
+        # If the manager had any arguments passed during the
+        # declaration (i.e: Manager()) then init it
+        # otherwise, it's a late bind.
+        if config is not None or session_options is not None \
+                or Model is not None or session_class is not None:
```

Comment:
I would say that we probably only want to look at `config` to determine if setup should proceed. While `Model` will typically be passed in, having `Model=None` is a supported use-case. The other arguments, `session_options` and `session_class` both have sane defaults and may not be provided.